From 21f32d866c8a88cd81479113a2e417fd0d040455 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 12 Mar 2020 19:05:26 +0200
Subject: [PATCH] [SW-10397] goya2: allow jtag access to security

Even with closed jtag interface we allow via range registers
access to PSOC_SECURITY over AXI so we can read the status
of image verification. DAP and all other PSOC will be closed.

Change-Id: I57939f59c87b356dfd08d6c1a43430ea81a12f5b
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index 8ab4618..5110338 100644
--- a/common/security.c
+++ b/common/security.c
@@ -19,14 +19,24 @@
 	/* override AXI prot bit by master IF to secure, unprivlaged access */
 	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
 		      mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+	/* enable access to PSOC_SECURITY only */
 	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
 	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
+	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
+
+	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
+		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
+		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
+		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
+		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
 }
 
 static void security_disable_smi_mstr_if(void)
@@ -36,14 +46,24 @@
 	/* override AXI prot bit by master IF to secure, unprivlaged access */
 	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
 		      mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+	/* enable access to PSOC_SECURITY only */
 	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
 	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
+	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
 		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
+
+	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
+		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
+		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
+		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
+		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
 }
 
 static void security_disable_jtag_mstr_if(void)
@@ -54,14 +74,24 @@
 	/* override AXI prot bit by master IF to secure, unprivlaged access */
 	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
 		      mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+	/* enable access to PSOC_SECURITY only */
 	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
 	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
+	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
+
+	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
+		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
+		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
+		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
+		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
 }
 
 static void security_enable_jtag_mstr_if(void)
